integrating-melodi-analytics/react-component-new.mdx

```diff
@@ -0,0 +1,173 @@
+   `npm install @melodi/melodi-sdk-react`
+
+2. Get a publishable API key from your Melodi Account.
+
+   - Log into your Melodi account and go to [settings](https://app.melodi.fyi/admin/settings).
+   - Click "Create publishable api key"
+   - Copy the newly created key
+
+3. Add the MelodiAuthProvider with your api key
+
+```jsx
+<MelodiAuthProvider apiKey="YOUR_API_KEY_HERE">
+  {children}
+</MelodiAuthProvider>
+```
+
+4. Add the MelodiFeedbackWidget and pass along the relevant info
+
+```jsx
+<MelodiFeedbackWidget
+  thread={{
+    messages: [
+      {
+        role: 'user',
+        content: userSubmittedText,
+        type: 'markdown'
+      },
+      {
+        role: 'assistant',
+        content: aiGeneratedText,
+        type: 'markdown'
+      }
+    ],
+    projectName: 'My Project',
+    metadata: {
+      environment: 'production',
+      feature: 'chat'
+    }
+  }}
+  userInfo={{
+    id: user.id,
+    email: user.email
+  }}
+/>
+```
+
+5. See feedback as it comes into your [Feedback Inbox](https://app.melodi.fyi/admin/feedback-inbox)
+
+## Usage Examples
+
+### Basic Usage
+```jsx
+<MelodiFeedbackWidget
+  thread={{
+    messages: [
+      {
+        role: 'user',
+        content: userSubmittedText,
+        type: 'markdown'
+      },
+      {
+        role: 'assistant',
+        content: aiGeneratedText,
+        type: 'markdown'
+      }
+    ],
+    projectName: 'My Project',
+    metadata: {
+      environment: 'production',
+      feature: 'chat'
+    }
+  }}
+  userInfo={{
+    id: user.id,
+    email: user.email
+  }}
+/>
+```
+
+### Advanced Usage with JSON Messages
+```jsx
+<MelodiFeedbackWidget
+  thread={{
+    messages: [
+      {
+        role: 'user',
+        content: 'Tell me a joke',
+        type: 'markdown'
+      },
+      {
+        role: 'json_tool',
+        type: 'json',
+        jsonContent: {
+          joke_setup: 'Why did the chicken cross the road?',
+          joke_punchline: 'To get to the other side!'
+        },
+        metadata: {
+          source: 'jokes_database'
+        }
+      },
+      {
+        role: 'assistant',
+        content: 'Why did the chicken cross the road? To get to the other side!',
+        type: 'markdown'
+      }
+    ],
+    projectName: 'Joke Bot',
+    externalId: 'joke-conversation-123',
+    metadata: {
+      environment: 'production',
+      feature: 'joke_generator',
+      team: 'content'
+    }
+  }}
+  userInfo={{
+    id: 'user-123',
+    email: 'user@example.com'
+  }}
+  companyName="My Company"
+/>
+```
+
+## Component Props
+
+### MelodiFeedbackWidget Props
+
+| Prop | Type | Required | Description |
+|------|------|----------|-------------|
+| `thread` | ThreadProps | Yes | Configuration for the conversation thread |
+| `userInfo` | UserInfo | No | Information about the current user |
+| `companyName` | string | No | Your company name to display in the feedback popover |
+
+### Message Types
+
+Messages in a thread can be one of three types:
```

Comment:
There are only 2 message types supported, "markdown" and "json"

The role on the message is a separate field and actually can be set to anything you want